### ISSUE TITLE
k8schain doesn't work with ECR

### ISSUE THREAD

**Issue by jonjohnsonjr** [google/go-containerregistry#355]
We import the aws credential provider package here: https://github.com/google/go-containerregistry/blob/1496eb6b0470a163639aaff730c61247df8ec0be/pkg/authn/k8schain/k8schain.go#L28

... expecting that to register ECR credential providers. That is not the case (there's no `init()` function in that package), so it doesn't work.

This works for the other credential provider packages, see gcp, azure, and rancher:
* https://github.com/kubernetes/kubernetes/blob/85f1205efa89008e6e94a92d66a9dd691cb59825/pkg/credentialprovider/gcp/metadata.go#L84
* https://github.com/kubernetes/kubernetes/blob/85f1205efa89008e6e94a92d66a9dd691cb59825/pkg/credentialprovider/azure/azure_credentials.go#L47
* https://github.com/kubernetes/kubernetes/blob/85f1205efa89008e6e94a92d66a9dd691cb59825/pkg/credentialprovider/rancher/rancher_registry_credentials.go#L60

Ideally we'd like to work automagically with ECR like we do with other platforms. I have [a proof of concept patch to k8schain](https://github.com/google/go-containerregistry/commit/5013c28cf3471e914d623c48223f02579293dc56), but we don't want to maintain this code. We want this to be fixed upstream in kubernetes/kubernetes and just consume that package like we do the others.

For inspiration, see:
https://github.com/kubernetes/kubernetes/blob/85f1205efa89008e6e94a92d66a9dd691cb59825/pkg/cloudprovider/providers/aws/aws.go#L1176-L1178

https://github.com/kubernetes/kubernetes/blob/85f1205efa89008e6e94a92d66a9dd691cb59825/pkg/cloudprovider/providers/aws/aws.go#L1265-L1268

https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/aws/regions.go

**Comment by Justin2997** [comment]
I did the modification on my dev environment and the fix seems to work for me.

**Comment by jonjohnsonjr** [comment]
Looks like someone attempted to do this, but from the linked issues they are looking at another approach:
https://github.com/kubernetes/kubernetes/pull/70675

**Comment by tiffanyfay** [comment]
@jonjohnsonjr Hey, are you on the Kubernetes slack or should I just FB message you? Not sure what "your change is related, but 'we need to be able to consume that more easily'" means.

**Comment by jonjohnsonjr** [comment]
The fix has been backported to 1.14, so once we update [our dependency on k/k](https://github.com/google/go-containerregistry/blob/31e00cede111067bae48bfc2cbfc522b0b36207f/Gopkg.toml#L30), we should be able to just close this.

**Comment by dprotaso** [comment]
@jonjohnsonjr which patch version of 1.14.x?

**Comment by dprotaso** [comment]
Also is there a way to support different k8s versions independently?

Or maybe rephrased - Can k8schain be versioned independently of ggcr and with k8s releases?

**Comment by jonjohnsonjr** [comment]
> @jonjohnsonjr which patch version of 1.14.x?

It's in v1.14.6: https://github.com/kubernetes/kubernetes/commit/ccdea031a6a610771917140380658d459437ee8e

> Or maybe rephrased - Can k8schain be versioned independently of ggcr and with k8s releases?

:man_shrugging:

**Comment by jonjohnsonjr** [comment]
Fixed in https://github.com/google/go-containerregistry/pull/649
